### PR TITLE
 Updated to reflect latest Clusterone & Tensorflow

### PR DESCRIPTION
**Pull request by adrianyi**
Replaced MNIST code to use recommended Estimator class.  Added the necessary TF_CONFIG snippet.
Replaced tf.app.flags with argparse as recommended by TF contributors (tf.app.flags is not officially supported).
Replaced data source, as this produced deprecation warnings for many previous TF versions and may disappear soon.
Rename tune.sh to gridsearch.sh.
Updated README & gridsearch.sh/gridsearch_distributed.sh to reflect changes in Clusterone CLI commands.

### CODE REVIEW

**Comment by adrianyi** [comment]
@MrUPGrade can you assign this to someone soon? I want to update the example as well, which depends on this. Thanks! :)

File: utils.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            raise ValueError(
+                'Invalid MNIST file %s: Expected 28x28 images, found %dx%d' %
+                (f.name, rows, cols))
+
+
+def check_labels_file_header(filename):
+    """Validate that filename corresponds to labels for the MNIST dataset."""
+    with tf.gfile.Open(filename, 'rb') as f:
+        magic = read32(f)
+        read32(f)  # num_items, unused
```
Comment (pkprzekwas) [inline utils.py]: Is this line needed? There is no assignment of the result.

File: mnist.py [hunk 2]
```diff
@@ -1,147 +1,226 @@
...
-      checkpoint_dir=FLAGS.log_dir) as sess:
-    while not sess.should_stop():
-      xs, ys = data.train.next_batch(FLAGS.batch_size, fake_data=False)
-      sess.run(train_op, feed_dict={images:xs, labels:ys})
+from utils import train_dataset, test_dataset
+
+
+def parse_args():
+    """Parse arguments"""
+    parser = ArgumentParser(formatter_class=ArgumentDefaultsHelpFormatter,
```
Comment (pkprzekwas) [inline mnist.py]: This part looks a bit dense. I would suggest adding some empty lines between different categories. I will make this more readable.

File: mnist.py [hunk 3]
```diff
@@ -1,147 +1,226 @@
...
+                                  local_root=opts.local_data_root,
+                                  local_repo='',
+                                  path=opts.data_subpath)
+    opts.log_dir = get_logs_path(root=opts.local_log_root)
+    opts.hidden_units = [int(n) for n in opts.hidden_units.split(',')]
+    if opts.worker_hosts:
+        opts.worker_hosts = opts.worker_hosts.split(',')
+    else:
+        opts.worker_hosts = []
+    if opts.ps_hosts:
```
Comment (pkprzekwas) [inline mnist.py]: I would add an empty line between these two if-else statements.

File: mnist.py [hunk 4]
```diff
@@ -1,147 +1,226 @@
...
+            tf.logging.warn('Expected job_name of worker or ps. Received {}.'.format(opts.job_name))
+        if not opts.ps_hosts:
+            tf.logging.warn('Expected ps_hosts, list of hostname:port pairs. Got {}. '.format(opts.ps_hosts) +
+                            'Example: --ps_hosts "localhost:2224" or --ps_hosts "localhost:2224,localhost:2225')
+        if not opts.worker_hosts:
+            tf.logging.warn('Expected worker_hosts, list of hostname:port pairs. Got {}. '.format(opts.worker_hosts) +
+                            'Example: --worker_hosts "localhost:2222,localhost:2223"')
+        tf.logging.warn('Ignoring distributed arguments. Running single mode.')
+        return {}
+    tf_config = {'task': {'type': opts.job_name, 'index': opts.task_index},
```
Comment (pkprzekwas) [inline mnist.py]: What do you think about this kind of dict formatting? I personally find it more readable.
```python    
tf_config = {
    'task': {
        'type': opts.job_name, 
        'index': opts.task_index
    },
    'cluster': {
        'master': [
            opts.worker_hosts[0]],
            'worker': opts.worker_hosts,
            'ps': opts.ps_hosts
    },
    'environment': 'cloud'
}
```

**Pull request merged**